### PR TITLE
Revamped Brakes, ABS and TC

### PR DESCRIPTION
**Pull request by ulteq**
Less code and:
- Almost completely removes brake induced wheel yitter
- Greatly improves braking on slopes ("Slopebrake")
- Greatly improves the anti lock system (ABS)
- Improves the traction control system (TC)

### CODE REVIEW

**Comment by only-a-ptr** [comment]
This is the kind of stuff you're better at. I'm always worried I'll subtly destabilize some mod I've never heard of.

**Comment by ulteq** [comment]
> How much testing do you do before you draw the conclusions?

Hours of testing sometimes days. But I've also worked on that part of the code before. That makes it a lot easier. There is one last issue, then it's ready for final review.

File: source/main/physics/BeamForcesEuler.cpp [hunk 1]
```diff
@@ -421,76 +423,58 @@ void Actor::calcForcesEulerCompute(int step, int num_steps)
...
                         brake_coef = 0.0f;
-                    }
-                }
-                else
-                {
-                    ar_wheels[i].firstLock = true;
                 }
 
-                if (ar_wheels[i].wh_speed > 0)
+                if (ar_wheels[i].wh_avg_speed > 0)
```
Comment (ulteq) [inline source/main/physics/BeamForcesEuler.cpp]: This is the key to why this new code works so much better.

File: source/main/physics/BeamForcesEuler.cpp [hunk 2]
```diff
@@ -401,6 +401,8 @@ void Actor::calcForcesEulerCompute(int step, int num_steps)
             total_torque = (m_num_axles == 0) ? engine_torque : intertorque[i];
         }
 
+        float wheel_slip = fabs(ar_wheels[i].wh_speed - curspeed) / std::max(1.0f, curspeed);
```
Comment (ulteq) [inline source/main/physics/BeamForcesEuler.cpp]: Our old wheel slip calculation was trash.

File: source/main/physics/BeamForcesEuler.cpp [hunk 3]
```diff
@@ -421,76 +423,58 @@ void Actor::calcForcesEulerCompute(int step, int num_steps)
...
-                        ar_wheels[i].wh_avg_speed = 0.0f;
-                        ar_wheels[i].firstLock = false;
-                    }
-                    // anti-jitter
-                    if (fabs(ar_wheels[i].wh_avg_speed) < 2.0f)
-                    {
-                        brake_coef = pow(fabs(ar_wheels[i].wh_speed), 2.0f);
-                    }
-                    else
+                    if (alb_pulse_state)
```
Comment (ulteq) [inline source/main/physics/BeamForcesEuler.cpp]: The old ABS pulse logic was trash.

**Pull request merged**